### ISSUE TITLE
Add Ability to List Rooms in Foreign¦Non-Local Homeserver Directories.

### ISSUE THREAD

**Issue by MilkManzJourDaddy** [turt2live/matrix-alias-bot#17]
This 'bot solves Room Admin's needing MXIDs on a bunch of Homeservers to get Aliases on them.  But any MXID local to such a Homeserver could currently do so on i.e. Riot-Web.  But listing a room in a Directory requires requisite Power Level that room Admin's might not want to give to unvetted users as they could i.e. set E2EE in the room.  And juggling multiple MXIDs just to manage rooms' listing on foreign (non-local) Homeservers' Directories is cumbersome, just as it would be solely for adding Aliases.  `↵↵¶`It seems logical to add this functionality to this particular 'bot until such time the spec' may, or may not, offer a streamlined solution.

**Comment by turt2live** [comment]
It's incredibly hard to parse what you're after, but I think you're looking for these two lines to be implemented?
https://github.com/turt2live/matrix-alias-bot/blob/master/src/CommandProcessor.ts#L45-L46

If yes, please note that the functionality was removed during the rewrite intentionally because I'm uncertain if it should be re-included or not.

**Comment by MilkManzJourDaddy** [comment]
Duplicate of #9

**Comment by turt2live** [comment]
It's not a duplicate if the partner issue is closed.

**Comment by MilkManzJourDaddy** [comment]
I see #9 has a commit [83b86f6](https://github.com/turt2live/matrix-alias-bot/commit/83b86f6b8829329eeb083b8744433c9427a29f87) ~9 Months ago, but do not see if it was executed.  The 'bot does not respond to `!room help`.  `↵↵¶`Btw, since it is the «Alias 'Bot», it would be easier to remember any help items if they were all included in `!alias help`.  `↵↵¶`Real world use-case: Betz is migrating hackerspace.be → hackerspaces.be and asked me to list rooms I created in the new directory, if the commit were in place a copy of the 'bot on hackerspaces.be could handle that.

**Comment by MilkManzJourDaddy** [comment]
`!room list` & `!room delist` versus `!room publish` & `!room unpublish` add to the confusion, but yes the functionality from https://github.com/turt2live/matrix-alias-bot/blob/master/src/CommandProcessor.ts#L45-L46  would be useful.  `↵↵¶`There are going to be a **LOT** of Homeservers, and 1 Homeserver Dev'Op' per Homeserver adding a copy of this 'bot will be much less work than many room Admin's getting MXIDs on many Homeservers, just to list¦publish in Directories and add Aliases.

**Comment by MilkManzJourDaddy** [comment]
Re-including the code and allowing those running this 'bot the ability to comment-out the ability to do this, makes logical sense.  This 'bot is a workaround for something that probably should be in the Matrix Spec', so let it work to the full advantage of those intending to use the thing.

**Comment by turt2live** [comment]
It hasn't been re-introduced because it wasn't used very often. The alias bot is one of the lowest priority projects of mine, so if the functionality is strongly desired then someone will have to write a PR for it at this time.

**Comment by MilkManzJourDaddy** [comment]
Thanks!
